src/main/java/com/isroot/stash/plugin/TimeLimitedMatcherFactory.java

```diff
@@ -0,0 +1,146 @@
+package com.isroot.stash.plugin;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A factory class used to generate Matcher instances that will throw if in use after their
+ * timeout.
+ */
```

Comment:
Can you add attribution to javadoc here? Maybe URL pointing to blog post code is from?

Reply:
Sure, I'll do it and update the PR.

Reply:
Done